db/migrate/20171213161035_add_indexes_on_images.rb

```diff
@@ -0,0 +1,23 @@
+class AddIndexesOnImages < ActiveRecord::Migration[5.0]
+  def up
+    deduped = Image.group(:name, :compute_resource_id, :operatingsystem_id).pluck('min(id)')
+    Image.where.not(id: deduped).each_with_index do |image, i|
+      say "Found images with duplicate name #{image.name} on Compute Resource #{image.compute_resource.name}!"
```

Comment:
You may need to wrap these in `User.as :admin` block because compute resources are taxable and no current user is set, so `image.compute_resource #=> nil`

Same for the lines below.

Reply:
this is a migration, so it may be run before any users are seeded. Changed method of looking it up, lets see if it works now.